### ISSUE TITLE
[BUG]: THIS RESULTED FROM USING ZENML WITH ML FLOW

### ISSUE THREAD

**Issue by 007ekho** [zenml-io/zenml#2540]
### Contact Details [Optional]

successiy007@gmail.com

### System Information

Failed to run step mlflow_model_deployer_step.
Failed to start service MLFlowDeploymentService[f10ad9d3-b7d3-4121-8183-58692fb18f04] (type: model-serving, flavor: mlflow)
  Administrative state: active
  Operational state: inactive
  Last status message: 'service daemon is not running'

### What happened?

Failed to run step mlflow_model_deployer_step.
Failed to start service MLFlowDeploymentService[f10ad9d3-b7d3-4121-8183-58692fb18f04] (type: model-serving, flavor: mlflow)
  Administrative state: active
  Operational state: inactive
  Last status message: 'service daemon is not running'

### Reproduction steps

1.import numpy as np
import pandas as pd
from zenml import pipeline, step
from zenml.config import DockerSettings
from zenml.constants import DEFAULT_SERVICE_START_STOP_TIMEOUT
from zenml.integrations.constants import MLFLOW
from zenml.integrations.mlflow.model_deployers.mlflow_model_deployer import (
    MLFlowModelDeployer,
)

from zenml.integrations.mlflow.services import MLFlowDeploymentService
from zenml.integrations.mlflow.steps import mlflow_model_deployer_step
from zenml.steps import BaseParameters, Output

from steps.cleaning_data import clean_df
from steps.evaluation import evaluate_model
from steps.ingest_data import ingest_df
from steps.model_train import train_model

docker_settings = DockerSettings(required_integrations= [MLFLOW])

class DeploymentTriggerConfig(BaseParameters):
    min_accuracy:float = 0.92

@step
def deployment_trigger(
    accuracy: float,
    config: DeploymentTriggerConfig,
):
    """implement a simple model deployment trigger that looks at the input model accuracy and decide to deploy or not"""
    return accuracy >= config.min_accuracy
@pipeline(enable_cache=False, settings = {"docker": docker_settings})
def continuous_deployment_pipeline(
    data_path: str,
    min_accuracy: float = 0.92,
    workers: int = 1,
    timeout: int =DEFAULT_SERVICE_START_STOP_TIMEOUT,):
    df = ingest_df(data_path=data_path)
    X_train, X_test, y_train, y_test = clean_df(df)
    model = train_model(X_train, X_test, y_train, y_test)
    mse,r2_score, rmse = evaluate_model(model, X_test,y_test)
    deployment_decision = deployment_trigger(rmse)
    mlflow_model_deployer_step(
        model= model,
        deploy_decision = deployment_decision,
        workers = workers,
        timeout = timeout
    )

    
2.
3.
...

### Relevant log output

```shell
Registered new version: (version 3).
Executing a new run.
Caching is disabled by default for continuous_deployment_pipeline.
Using user: default
Using stack: mlflow_stack
  artifact_store: default
  experiment_tracker: mlflow_tracker
  orchestrator: default
  model_deployer: mlflow
Step ingest_df has started.
ingesting data from C:/Users/USER/Downloads/customer_satisfaction/data/olist_customers_dataset.csv 
Daemon functionality is currently not supported on Windows.
Step ingest_df has finished in 4.496s.
Step clean_df has started.
C:\Users\USER\Downloads\customer_satisfaction\src\data_cleaning.py:37: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  data["product_weight_g"].fillna(data["product_weight_g"].median(), inplace = True)
C:\Users\USER\Downloads\customer_satisfaction\src\data_cleaning.py:38: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  data["product_length_cm"].fillna(data["product_length_cm"].median(), inplace = True)
C:\Users\USER\Downloads\customer_satisfaction\src\data_cleaning.py:39: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  data["product_height_cm"].fillna(data["product_height_cm"].median(), inplace = True)
C:\Users\USER\Downloads\customer_satisfaction\src\data_cleaning.py:40: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  data["product_width_cm"].fillna(data["product_width_cm"].median(), inplace = True)
Data cleaning completed
Daemon functionality is currently not supported on Windows.
Step clean_df has finished in 3.073s.
Step train_model has started.
2024/03/18 15:59:10 WARNING mlflow.utils.autologging_utils: You are using an unsupported version of sklearn. If you encounter errors during autologging, try upgrading / downgrading sklearn to a supported version, or try upgrading MLflow.
2024/03/18 15:59:12 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "C:\Users\USER\Downloads\customer_satisfaction\.pipeml\lib\site-packages\mlflow\data\digest_utils.py:29: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead."        
2024/03/18 15:59:13 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "C:\Users\USER\Downloads\customer_satisfaction\.pipeml\lib\site-packages\mlflow\types\utils.py:393: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."
2024/03/18 15:59:13 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "C:\Users\USER\Downloads\customer_satisfaction\.pipeml\lib\site-packages\mlflow\types\utils.py:393: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."
Model training completed
Daemon functionality is currently not supported on Windows.
C:\Users\USER\Downloads\customer_satisfaction\.pipeml\lib\site-packages\zenml\integrations\mlflow\experiment_trackers\mlflow_experiment_tracker.py:254: FutureWarning: ``mlflow.gluon.autolog`` is deprecated since 2.5.0. This method will be removed in a future release.
  module.autolog(disable=True)
Step train_model has finished in 10.468s.
Step evaluate_model has started.
calculating MSE
MSE: 1.8640770533975461
calculating R2_score
R2: 0.017729030402296564
calculating RMSE
C:\Users\USER\Downloads\customer_satisfaction\.pipeml\lib\site-packages\sklearn\metrics\_regression.py:483: FutureWarning: 'squared' is deprecated in version 1.4 and will be removed in 1.6. To calculate the root mean squared error, use the function'root_mean_squared_error'.
  warnings.warn(
R2: 1.365312071798073
Daemon functionality is currently not supported on Windows.
C:\Users\USER\Downloads\customer_satisfaction\.pipeml\lib\site-packages\zenml\integrations\mlflow\experiment_trackers\mlflow_experiment_tracker.py:254: FutureWarning: ``mlflow.gluon.autolog`` is deprecated since 2.5.0. This method will be removed in a future release.
  module.autolog(disable=True)
Step evaluate_model has finished in 0.757s.
Step deployment_trigger has started.
Daemon functionality is currently not supported on Windows.
Step deployment_trigger has finished in 0.109s.
Caching disabled explicitly for mlflow_model_deployer_step.
Step mlflow_model_deployer_step has started.
Daemon functionality is currently not supported on Windows.
Daemon functionality is currently not supported on Windows.
Calling stop method...
Daemon functionality is currently not supported on Windows.
Daemon functionality is currently not supported on Windows.
Daemon functionality is currently not supported on Windows.
stop method executed successfully.                                                                 
Updating an existing MLflow deployment service: MLFlowDeploymentService[f10ad9d3-b7d3-4121-8183-58692fb18f04] (type: model-serving, flavor: mlflow)
Calling stop method...
Daemon functionality is currently not supported on Windows.
Daemon functionality is currently not supported on Windows.
Daemon functionality is currently not supported on Windows.
stop method executed successfully.                                                                 
Calling start method...                                                                            
Daemon functionality is currently not supported on Windows.
Daemon functionality is currently not supported on Windows.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
